--- a/src/com/mojang/minecraft/server/HeartbeatManager.java
+++ b/src/com/mojang/minecraft/server/HeartbeatManager.java
@@ -1,11 +1,6 @@
 package com.mojang.minecraft.server;
 
-import com.mojang.minecraft.server.MinecraftServer;
-import java.io.BufferedReader;
-import java.io.DataOutputStream;
-import java.io.FileWriter;
-import java.io.InputStreamReader;
-import java.io.PrintWriter;
+import java.io.*;
 import java.net.HttpURLConnection;
 import java.net.URL;
 
@@ -18,16 +13,16 @@
 
 
    HeartbeatManager(MinecraftServer var1, String var2) {
+      super();
       this.server = var1;
       this.name = var2;
-      super();
    }
 
    public final void run() {
       HttpURLConnection var1 = null;
 
       try {
-         (var1 = (HttpURLConnection)(new URL("http://www.minecraft.net/heartbeat.jsp")).openConnection()).setRequestMethod("POST");
+         (var1 = (HttpURLConnection)(new URL("http://www.classicube.net/heartbeat.jsp")).openConnection()).setRequestMethod("POST");
          var1.setDoOutput(true);
          var1.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
          var1.setRequestProperty("Content-Length", "" + Integer.toString(this.name.getBytes().length));
