--- a/src/com/mojang/minecraft/server/PlayerManager.java
+++ b/src/com/mojang/minecraft/server/PlayerManager.java
@@ -1,12 +1,6 @@
 package com.mojang.minecraft.server;
 
-import com.mojang.minecraft.server.MinecraftServer;
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.FileReader;
-import java.io.FileWriter;
-import java.io.IOException;
-import java.io.PrintWriter;
+import java.io.*;
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Set;
