--- a/src/com/mojang/minecraft/level/BlockMap.java
+++ b/src/com/mojang/minecraft/level/BlockMap.java
@@ -1,11 +1,10 @@
 package com.mojang.minecraft.level;
 
 import com.mojang.minecraft.Entity;
-import com.mojang.minecraft.level.BlockMap$Slot;
-import com.mojang.minecraft.level.UNKNOWN1;
 import com.mojang.minecraft.model.Vector3D;
 import com.mojang.minecraft.phys.AABB;
 import com.mojang.minecraft.render.Frustrum;
+
 import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.List;
