--- a/src/com/mojang/minecraft/level/Level.java
+++ b/src/com/mojang/minecraft/level/Level.java
@@ -168,8 +168,6 @@
       this.listeners.add(var1);
    }
 
-   public void finalize() {}
-
    public void removeListener$74652038(MinecraftServer var1) {
       this.listeners.remove(var1);
    }
@@ -318,7 +316,7 @@
    }
 
    public boolean isLit(int var1, int var2, int var3) {
-      return var1 >= 0 && var2 >= 0 && var3 >= 0 && var1 < this.width && var2 < this.depth && var3 < this.height?var2 >= this.blockers[var1 + var3 * this.width]:true;
+      return var1 < 0 || var2 < 0 || var3 < 0 || var1 >= this.width || var2 >= this.depth || var3 >= this.height || var2 >= this.blockers[var1 + var3 * this.width];
    }
 
    public int getTile(int var1, int var2, int var3) {
@@ -327,7 +325,7 @@
 
    public boolean isSolidTile(int var1, int var2, int var3) {
       Block var4;
-      return (var4 = Block.blocks[this.getTile(var1, var2, var3)]) == null?false:var4.isSolid();
+      return (var4 = Block.blocks[this.getTile(var1, var2, var3)]) != null && var4.isSolid();
    }
 
    public void tickEntities() {
