--- a/src/com/mojang/minecraft/level/tile/FlowerBlock.java
+++ b/src/com/mojang/minecraft/level/tile/FlowerBlock.java
@@ -1,8 +1,9 @@
 package com.mojang.minecraft.level.tile;
 
+import com.fukkit.API;
 import com.mojang.minecraft.level.Level;
-import com.mojang.minecraft.level.tile.Block;
 import com.mojang.minecraft.phys.AABB;
+
 import java.util.Random;
 
 public class FlowerBlock extends Block {
@@ -14,7 +15,16 @@
       this.setBounds(0.5F - var3, 0.0F, 0.5F - var3, var3 + 0.5F, var3 * 3.0F, var3 + 0.5F);
    }
 
+   private static boolean shouldTick()
+   {
+      return Boolean.parseBoolean(API.getInstance().getServer().getProperties().getProperty("tick-flowers"));
+   }
+
    public void update(Level var1, int var2, int var3, int var4, Random var5) {
+      if (!shouldTick())
+      {
+         return;
+      }
       int var6 = var1.getTile(var2, var3 - 1, var4);
       if(!var1.isLit(var2, var3, var4) || var6 != Block.DIRT.id && var6 != Block.GRASS.id) {
          var1.setTile(var2, var3, var4, 0);
