--- a/src/com/mojang/minecraft/level/tile/MushroomBlock.java
+++ b/src/com/mojang/minecraft/level/tile/MushroomBlock.java
@@ -1,8 +1,7 @@
 package com.mojang.minecraft.level.tile;
 
 import com.mojang.minecraft.level.Level;
-import com.mojang.minecraft.level.tile.Block;
-import com.mojang.minecraft.level.tile.FlowerBlock;
+
 import java.util.Random;
 
 public final class MushroomBlock extends FlowerBlock {
